src/algebra/ordered_ring.lean

```diff
@@ -655,6 +655,11 @@ lemma neg_of_mul_pos_right (h : 0 < a * b) (ha : a ≤ 0) : b < 0 :=
 by haveI := @linear_order.decidable_le α _; exact
 lt_of_not_ge (λ hb, absurd h (decidable.mul_nonpos_of_nonpos_of_nonneg ha hb).not_lt)
 
+@[priority 100] -- see Note [lower instance priority]
+instance linear_ordered_semiring.to_linear_ordered_add_comm_monoid {α : Type*}
+  [h : linear_ordered_semiring α] : linear_ordered_add_comm_monoid α :=
+{ ..h }
```

Comment:
What happens if you change line 417 to:
```lean
@[protect_proj]
class linear_ordered_semiring (α : Type u) extends
  ordered_semiring α, linear_ordered_add_comm_monoid α, nontrivial α
```
instead? That avoids having a shortcut path to `linear_order`, which is probably a good thing